core/trino-main/src/main/java/io/trino/metadata/TableExecuteHandle.java

```diff
@@ -0,0 +1,103 @@
+ */
+package io.trino.metadata;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.trino.connector.CatalogName;
+import io.trino.spi.connector.ConnectorTableExecuteHandle;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public final class TableExecuteHandle
+{
+    private final CatalogName catalogName;
+    private final ConnectorTransactionHandle transactionHandle;
+    private final ConnectorTableExecuteHandle connectorHandle;
+    private final TableHandle sourceTableHandle;
+
+    @JsonCreator
+    public TableExecuteHandle(
+            @JsonProperty("catalogName") CatalogName catalogName,
+            @JsonProperty("transactionHandle") ConnectorTransactionHandle transactionHandle,
+            @JsonProperty("connectorHandle") ConnectorTableExecuteHandle connectorHandle)
+    {
+        this.catalogName = requireNonNull(catalogName, "catalogName is null");
+        this.transactionHandle = requireNonNull(transactionHandle, "transactionHandle is null");
+        this.connectorHandle = requireNonNull(connectorHandle, "connectorHandle is null");
+        this.sourceTableHandle = new TableHandle(catalogName, connectorHandle.getSourceTableHandle(), transactionHandle, Optional.empty());
+    }
+
+    @JsonProperty
+    public CatalogName getCatalogName()
+    {
+        return catalogName;
+    }
+
+    @JsonProperty
+    public ConnectorTransactionHandle getTransactionHandle()
+    {
+        return transactionHandle;
+    }
+
+    @JsonProperty
+    public ConnectorTableExecuteHandle getConnectorHandle()
+    {
+        return connectorHandle;
+    }
+
+    @JsonIgnore
+    public TableHandle getSourceTableHandle()
+    {
+        return sourceTableHandle;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(catalogName, transactionHandle, connectorHandle, sourceTableHandle);
+    }
+
+    public TableExecuteHandle withConnectorHandle(ConnectorTableExecuteHandle connectorHandle)
+    {
+        return new TableExecuteHandle(catalogName, transactionHandle, connectorHandle);
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        TableExecuteHandle o = (TableExecuteHandle) obj;
+        return Objects.equals(this.catalogName, o.catalogName) &&
+                Objects.equals(this.transactionHandle, o.transactionHandle) &&
+                Objects.equals(this.connectorHandle, o.connectorHandle) &&
+                Objects.equals(this.sourceTableHandle, o.sourceTableHandle);
+    }
+
+    @Override
+    public String toString()
+    {
+        return catalogName + ":" + connectorHandle;
```

Comment:
add "Execute"

Reply:
Can do but it should be part of `connectorHandle` already. 
We are not prefixing for `InsertTH`